### PR TITLE
I finished task 4

### PR DESCRIPTION
**Pull request by XingWei0920**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -47,4 +47,49 @@ describe("/api/topics", () => {
...
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: date.toISOString(),
+          votes: 100,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        }));
+        
+        response.body.article.forEach((article) => {
```
Comment (ghost) [inline __tests__/app.test.js]: This is not necessary at all as you are asserting the data types in the test above.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -47,4 +47,49 @@ describe("/api/topics", () => {
...
+  });
+
+  test("GET:404 sends an NOT FOUND error if the article_id does not exist", () => {
+    return request(app)
+      .get("/api/articles/20")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("not found");
+      });
+  });
```
Comment (ghost) [inline __tests__/app.test.js:95]: What if the article id is not a number? What would happen if I searched for /api/articles/banana and what status code would that be?

File: models/server-models.js [hunk 3]
```diff
@@ -1,10 +1,25 @@
...
     let queryString =
     "SELECT * FROM topics;";
 
     return db.query(queryString).then((result) => {
         return result.rows;
       });
+}
+
+exports.selectArticleById= (article_id) => {
+    let queryString =
```
Comment (ghost) [inline models/server-models.js]: Is this query string being changed at any point? If not, this should be declared with const.

**Review by ghost — CHANGES_REQUESTED** [review]
Almost there on this Xing, just missing a test. If you could check that and resubmit when you're ready please.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -47,4 +47,46 @@ describe("/api/topics", () => {
...
+      .then((response) => {
+      const date = new Date('2020-07-09 21:11:00') 
+        expect(response.body.article.length).toBe(1);
+        expect(response.body.article[0]).toEqual(expect.objectContaining({
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: date.toISOString(),
```
Comment (Slyose) [inline __tests__/app.test.js:80]: Can also just expect.any(String) here, meaning you wont need to make this date object and convert it to a string

**Review by Slyose — APPROVED** [review]
Good response to feedback! Feel free to merge ✅

(also left a note about how you're testing for created_at, but it's not a dealbreaker)

**Pull request merged**